src/frontends/tensorflow/src/saved_model.hpp

```diff
@@ -0,0 +1,240 @@
+    bool get_variable(const std::string& name, char** data, size_t* size) {
+        auto varItem = m_variables_index.find(name);
+        if (varItem == m_variables_index.end()) {
+            return false;
+        }
+        if (data != nullptr) {
+            *data = varItem->second.data();
+        }
+        if (size != nullptr) {
+            *size = varItem->second.size();
+        }
+        return true;
+    }
+
+    /// \brief Returns data and size of data of mapped variable from trackable object graph to variables index
+    /// \param name Name of a mapping variable
+    /// \param data Pointer on a pointer where data pointer will be returned
+    /// \param size Pointer on a variable which will stores data size
+    /// \returns Returns true in case variable was found, false otherwise (data and size will be untouched)
+    bool get_mapped_variable(const std::string& name, char** data, size_t* size) {
+        auto mapItem = m_variables_map.find(name);
+        if (mapItem == m_variables_map.end()) {
+            return false;
+        }
+        return get_variable(mapItem->second, data, size);
+    }
+
+    /// \brief Checks if variable has a mapped pair
+    /// \param name Name of variable for checking existance
+    /// \returns True in case variable has mapped value and false otherwise
+    bool has_mapped_variable(const std::string& name) {
+        auto mapItem = m_variables_map.find(name);
+        return mapItem != m_variables_map.end();
+    }
+
+    /// \brief Returns shared pointer to a requested shard_id, or nullptr in case of shard_id isn't found
+    /// \param shard_id Requested shard_id
+    /// \returns Valid shared_ptr with ifstream or with nullptr if shard isn't found
+    std::shared_ptr<std::ifstream> get_data_file(const int32_t shard_id) {
+        auto result = m_data_files.find(shard_id);
+        return result != m_data_files.end() ? result->second : nullptr;
+    }
+
+    /// \brief Adds variable mapping to the variables map
+    /// \param var_name Variable full name (from .index file)
+    /// \param map_name Mapped name
+    /// \param rewrite Rewrite mapped value in case it exists
+    /// \returns True if map updated. False if nothing changed (if variable exists and rewrite is false).
+    bool map_variable(const std::string& var_name, const std::string& map_name, bool rewrite = false) {
+        if (m_variables_map.find(var_name) != m_variables_map.end() && rewrite == false) {
+            return false;
+        }
+
+        m_variables_map[var_name] = map_name;
+        return true;
+    }
+
+private:
+    // Internal implementation of saved model reading
+    void read_variables_index_block(std::ifstream& fs,
+                                    const VIBlock* index,
+                                    std::vector<char>& data,
+                                    uint32_t* offset,
+                                    uint32_t* offset_end);
```

Comment:
propose to use reference instead of pointers.
```suggestion
                                    uint32_t& offset,
                                    uint32_t& offset_end);
```

Reply:
It is doable, but code become unreadable because looking on function call you cannot understand it might be changed inside.
But I have no problem with that, most of code is unreadable without powerful IDE.